src/canmatrix/formats/dbc.py

```diff
@@ -450,6 +450,20 @@ class FollowUps(object):
     frame = None
     boardUnit = None
     db = canmatrix.CanMatrix()
+    framesById = {}
+    
+    def hash_arbitration_id(arbitration_id):
+        return hash((arbitration_id.id, arbitration_id.extended))
+
+    def get_frame_by_id(arbitration_id):
+        try:
+            return framesById[hash_arbitration_id(arbitration_id)]
+        except KeyError:
+            return None
+            
+    def add_frame_by_id(frame):
+        framesById[hash_arbitration_id(frame.arbitration_id)] = frame
+
```

Comment:
If we don't change the core library to work 'this way' then for a just-import solution I would tend to make a little class (in global scope) I think instead of on-the-fly functions with closures.

```python3
def hash_arbitration_id(...  # as already submitted

@attr.s
class IdFrameMap:
    frames_by_id = attr.ib(factory=dict)

    def get(self, id):
        return self.frames_by_id[hash_arbitration_id(id)]

    def set(self, id, frame):
        self.frames_by_id[hash_arbitration_id(id)] = frame
```

I considered using `__getitem__`/`__setitem__` but since this is really not a proper dict per-se I sort of tend to want it to look a little different in its use.  But, that's a mild tendency at this point, not a hard preference.